test/correctness/allocator_alignment.cpp

```diff
@@ -0,0 +1,104 @@
+#include "Halide.h"
+#include "halide_benchmark.h"
+#include <stdio.h>
+
+using namespace Halide;
+using namespace Halide::ConciseCasts;
+
+namespace {
+
+void check(int r) {
+    assert(r == 0);
+}
+
+void run_test(Target t) {
+    // Must call this so that the changes in cached runtime are noticed
+    Halide::Internal::JITSharedRuntime::release_all();
+
+    class TestGen1 : public Generator<TestGen1> {
+    public:
+        Input<Buffer<uint32_t, 2>> img_{"img"};
+        Input<uint32_t> offset_{"offset"};
+        Output<Buffer<uint32_t, 2>> out_{"out"};
+
+        void generate() {
+            Var x("x"), y("y");
+
+            // Make a copy so that halide_malloc() is called by the generated
+            // code (since Halide::Runtime::Buffer doesn't use halid_malloc(),
+            // see https://github.com/halide/Halide/issues/7188).
+            Func copy("copy");
+            copy(x, y) = img_(x, y);
+
+            out_(x, y) = u32_sat(copy(x, y) + offset_);
+
+            copy.compute_root().store_in(MemoryType::Heap).vectorize(x, natural_vector_size<uint32_t>());
+            out_.vectorize(x, natural_vector_size<uint32_t>());
+        }
+    };
+
+    // Make it large enough so that we won't attempt to use stack instead of heap
+    const int W = t.natural_vector_size<uint32_t>() * 256;
+    const int H = 2048;
+
+    Buffer<uint32_t> in1(W, H);
+    Buffer<uint32_t> in2(W, H);
+
+    for (int i = 0; i < W; i++) {
+        for (int j = 0; j < H; j++) {
+            in1(i, j) = i + j * 10;
+            in2(i, j) = i * 10 + j;
+        }
+    }
+
+    const GeneratorContext context(t);
+
+    auto gen = TestGen1::create(context);
+    Callable c = gen->compile_to_callable();
+
+    const uint32_t offset1 = 42;
+    Buffer<uint32_t> out1(W, H);
+    check(c(in1, offset1, out1));
```

Comment:
Can Callables return non-zero without a custom error handler? I thought we just threw an exception.

Reply:
Probably not, but it's good hygiene.